### ISSUE TITLE
bug: Might run into infinite loop if there's no available suggestion.

### ISSUE THREAD

**Issue by moong23** [cdhawke/accessible-colors#4]
Hi 😄

I found your repository while searching for "Visible Color contrasting with the background color."

In your code, I noticed that the while loop will never break if there are no color suggestions available. For example, `getRandomAAAColor('#888888', false);`
will never be able to break out of the while loop, which might trigger an infinite loop.

So, here's my suggestion to prevent the issue:

```
/**
 * getRandomAAAColor will return a random color that is accessible based on the
 * WCAG 2.0 AAA standard, which requires a contrast ratio of at least 7:1. It will
 * take into account the luminance of the background color (hash).
 * @param background - the background color to use for the contrast ratio calculation.
 * @param large - whether the text should be considered large, adjusting the contrast ratio requirement to 4.5:1.
 * @param maxAttempts - the maximum number of attempts to find a suitable color.
 * @returns a random color that is accessible based on the WCAG 2.0 AAA standard or `null` if no suitable color is found.
 */
export const getRandomAAAColor = (
  background: string,
  large = false,
  maxAttempts = 1000
): string | null => {
  let attempts = 0;
  let color = randomColor();
  
  while (!isAAAContrast(background, color, large)) {
    if (attempts >= maxAttempts) {
      return null; // Return null if no suitable color is found within the maxAttempts limit
    }
    color = randomColor();
    attempts++;
  }

  return color;
};
```

**Comment by cdhawke** [comment]
Thanks @moong23! Let me take a look, and thanks for the suggestion. I think this function should always be able to find a complement, so maybe something broke when I updated dependencies recently.

**Comment by cdhawke** [comment]
@moong23 - this should be fixed - I just went with your suggestion so thank you! I misunderstood the function that was having issues originally thinking it was the binary search version.
